src/scheduler/agent/database.c

```diff
@@ -1132,6 +1118,30 @@ char* get_email_command(scheduler_t* scheduler, char* user_email)
         g_string_append(client_cmd, "warn");
       }
     }
+    if(g_hash_table_contains(smtpvariables, "SMTPAuthUser") && g_hash_table_contains(smtpvariables, "SMTPAuthPasswd")
+          && g_hash_table_contains(smtpvariables, "SMTPHostName") && g_hash_table_contains(smtpvariables, "SMTPPort"))
```

Comment:
This will result in some issue. For example, with current implementation, if I do not provide `SMTPAuthPasswd` (as my server do not required one), I will never receive an email. With the current implementation, that is not the case.

I would suggest to modify current code and build up a server string with the existing checks (if password is not provided, don't add it to the string). And then use the string in the command as value for `-S mta=\"%s\"`.